test/functional/provider/admin/messages/inbox_controller_test.rb

```diff
@@ -48,4 +48,24 @@ def test_index_system_message
     assert_select 'title', "Inbox - Index | Red Hat 3scale API Management"
     assert_select '#export-to-csv', false, 'Export all Messages'
   end
+
+  test 'Prevent sending reply without subject' do
+    login_as(@admin)
+    reply = @message.reply
+    reply.update_attributes(body: 'Reply Body', subject: nil)
+
+    MessageWorker.drain
+    assert msg = Message.last
+    assert_not_equal "Reply Body", msg.body
```

Comment:
Shouldn't it better to assert `message.valid?`

Reply:
Sure we can check that but why you think that this is not better , this test also check what we want to check